man/machine-id.xml

```diff
@@ -53,18 +53,26 @@
   <refsect1>
     <title>Description</title>
 
-    <para>The <filename>/etc/machine-id</filename> file contains the unique machine ID of the local
-    system that is set during installation. The machine ID is a single newline-terminated,
-    hexadecimal, 32-character, lowercase ID. When decoded from hexadecimal, this corresponds to a
-    16-byte/128-bit value.</para>
-
-    <para>The machine ID is usually generated from a random source
-    during system installation and stays constant for all subsequent
-    boots. Optionally, for stateless systems, it is generated during
-    runtime at early boot if it is found to be empty.</para>
-
-    <para>The machine ID does not change based on local or network configuration or when hardware is
-    replaced. Due to this and its greater length, it is a more useful replacement for the
+    <para>The <filename>/etc/machine-id</filename> file contains the unique machine ID of
+    the local system that is set during installation or boot. The machine ID is a single
+    newline-terminated, hexadecimal, 32-character, lowercase ID. When decoded from
+    hexadecimal, this corresponds to a 16-byte/128-bit value. This ID may not be all
+    zeros.</para>
+
+    <para>The machine ID is usually generated from a random source during system
+    installation or first boot and stays constant for all subsequent boots. Optionally,
+    for stateless systems, it is generated during runtime during early boot if necessary.
+    </para>
+
+    <para>The machine ID may be set, for example when network booting, with the
+    <varname>systemd.machine_id=</varname> kernel command line parameter or by passing the
+    option <option>--machine-id=</option> to systemd. An ID is specified in this manner
+    has higher priority and will be used instead of the ID stored in
+    <filename>/etc/machine-id</filename>.</para>
```

Comment:
is that really the case? i remember we used such passed-in uuids only when we /etc/machin-id was so far unset

Reply:
Yes, I just tested, and it even gets committed to /etc/machine-id.